auction9-backend/handler.js

```diff
@@ -171,15 +171,76 @@ export const stopActiveAuction = async (event, context) => {
  * GET: /wonauctions
  */
 export const getUserWonAuctions = async (event, context) => {
-   try {
-     let currentUserId = event.multiValueQueryStringParameters.userId[0];
-     let resultsUserWonAuctions = await mysql.query(`SELECT * FROM tbl_auction WHERE winner=?`, [currentUserId]);
-     await mysql.end();
-     return generateResponse(200, resultsUserWonAuctions);
-   } catch (error) {
-     console.log(error);
-     return generateResponse(400, {
-       message: 'There was an error while getting my won auctions.'
-     });
-   }
+  try {
+    let currentUserId = event.multiValueQueryStringParameters.userId[0];
+    let resultsUserWonAuctions = await mysql.query(`SELECT * FROM tbl_auction WHERE winner=?`, [currentUserId]);
+    await mysql.end();
+    return generateResponse(200, resultsUserWonAuctions);
+  } catch (error) {
+    console.log(error);
+    return generateResponse(400, {
+      message: 'There was an error while getting my won auctions.'
+    });
+  }
+};
+
+
+/* Allowed status changes:
+ * order status: INACTIVE -> ACTIVE -> FINISHED -> REALIZED
+ * current -> new :
+ * INACTIVE -> ACTIVE   [x]
+ * ACTIVE -> INACTIVE   [x]
+ * ACTIVE -> FINISHED   [x]
+ * FINISHED -> REALIZED [x]
+ */
+/* realizeFinishedAuction - will update status for auction to 'realized'
+ * PUT: /myauctions/id
+ */
+export const realizeFinishedAuction = async (event, context) => {
```

Comment:
Be aware that status logic should be also supported in 'edit auction lambda'.
If you move it there, we can use only one method.

Reply:
Yes correct, but at this time Dragan was working on 'edit' part